src/cup1d/nuisance/mean_flux_model.py

```diff
@@ -2,129 +2,145 @@
 import copy
 import os
 from scipy.interpolate import interp1d
-
-import lace
 from cup1d.likelihood import likelihood_parameter
 
 
 class MeanFluxModel(object):
     """Use a handful of parameters to model the mean transmitted flux fraction
-        (or mean flux) as a function of redshift. 
-         For now, we use a polynomial to describe log(tau_eff) around z_tau.
-         """
-
-    def __init__(self,z_tau=3.0,ln_tau_coeff=None,fid_fname=None,
-                free_param_names=None):
+    (or mean flux) as a function of redshift.
+     For now, we use a polynomial to describe log(tau_eff) around z_tau.
+    """
+
+    def __init__(
+        self,
+        z_tau=3.0,
+        ln_tau_coeff=None,
+        free_param_names=None,
+        fid_igm=None,
+    ):
         """Construct model as a rescaling around a fiducial mean flux"""
 
-        # figure out filename
-        if not fid_fname:
-            basedir="/data/sim_suites/Australia20/"
-            repo=f"{lace.__path__[0]}"
-            fid_fname="{}/{}/fiducial_igm_evolution.txt".format(repo,basedir)
+        if fid_igm is None:
+            fname = (
+                os.environ["LACE_REPO"]
+                + "/src/lace/data/sim_suites/Australia20/IGM_histories.npy"
+            )
+            try:
+                igm_hist = np.load(fname, allow_pickle=True).item()
+            except:
+                raise ValueError(
+                    fname
+                    + "not found. You can produce it using LaCE"
+                    + r"\n script save_"
+                    + sim_igm[:3]
+                    + "_IGM.py"
+                )
+            fid_igm = igm_hist["mpg_central"]
+
+        _ = np.argwhere(fid_igm["tau_eff"] != 0)[:, 0]
```

Comment:
Shouldn't the code crash (or at least print a warning) if one of the values was 0?

Reply:
Sure, I will add it. As I mentioned yesterday, it turns out that this is somewhat common for Nyx